samples/web/content/testrtc/js/mictest.js

```diff
@@ -31,7 +31,11 @@ function MicTest() {
 
 MicTest.prototype = {
   run: function() {
-    doGetUserMedia(this.constraints, this.gotStream.bind(this));
+    if (typeof audioContext === 'undefined') {
+      reportFatal('WebAudio is not supported, test cannot run.');
```

Comment:
Can you convert the reportFatal into "reportError" and fix the testFinished() code paths?